src/main/java/in/twizmwaz/cardinal/module/modules/stats/Stats.java

```diff
@@ -0,0 +1,151 @@
+        for (Map.Entry<Player, Integer> e : count.entrySet()) {
+            if (mostRepeted == null || mostRepeted.getValue() < e.getValue()) {
+                mostRepeted = e;
+            }
+        }
+        return mostRepeted != null ? mostRepeted.getKey() : null;
+    }
+
+    private int getKillsByPlayer(Player player) {
+        int kills = 0;
+        for (Tracker tracker : stats) {
+            if (tracker.getDamager().equals(player)) kills++;
+        }
+        return kills;
+    }
+
+    private int getDeathsByPlayer(Player player) {
+        int deaths = 0;
+        for (Tracker tracker : stats) {
+            if (tracker.getDamaged().equals(player)) deaths++;
+        }
+        return deaths;
+    }
+
+
+    @EventHandler
+    public void onPlayerDeath(PlayerDeathEvent event) {
+        if (event.getEntity() != null && event.getEntity().getKiller() != null) {
+            add(new Tracker(event.getEntity(), event.getEntity().getKiller(), event.getEntity().getKiller().getItemInHand() != null ? event.getEntity().getKiller().getItemInHand() : null));
+        }
+    }
+
+    @EventHandler
+    public void matchEndEvent(MatchEndEvent event) {
+        if (GameHandler.getGameHandler().getPlugin().getConfig().getBoolean("broadcastStats")) {
+            for (Tracker tracker : stats) {
+                totalDeaths++;
+                totalKills++;
+            }
+            for (Player player : Bukkit.getOnlinePlayers()) {
+                player.sendMessage("");
+                player.sendMessage(ChatColor.GOLD + "" + ChatColor.STRIKETHROUGH + "---------------" + ChatColor.AQUA + " Stats " + ChatColor.GOLD + "" + ChatColor.STRIKETHROUGH + "---------------");
+                player.sendMessage("              " + ChatColor.GREEN + "Kills: " + totalKills + "              " + ChatColor.RED + "Deaths: " + totalDeaths);
+                for (TeamModule team : TeamUtils.getTeams()) {
+                    player.sendMessage("");
+                    if (team != null && !team.isObserver()) {
+                        int teamKills = 0;
+                        int teamDeaths = 0;
+                        for (Tracker tracker : stats) {
+                            if (TeamUtils.getTeamByPlayer(tracker.getDamager()).equals(team)) teamKills++;
+                            if (TeamUtils.getTeamByPlayer(tracker.getDamaged()).equals(team)) teamDeaths++;
+                        }
+                        if (getPlayerWithMostKillsOnTeam(team) != null && getPlayerWithMostDeathsOnTeam(team) != null) {
+                            player.sendMessage("    " + team.getColor() + ChatColor.BOLD + team.getCompleteName() + ChatColor.GREEN + "    kills: " + ChatColor.WHITE + teamKills + ChatColor.RED + "    deaths     " + ChatColor.WHITE + teamDeaths + ChatColor.GOLD + " KD " + Math.round((double) teamKills / ((double) teamDeaths > 0 ? (double) teamDeaths : 1.0) * 100) / 100 );
+                            player.sendMessage("");
```

Comment:
Does it actually? I'm pretty sure the Minecraft Client won't print a completely blank line.